src/DynamoCore/Search/SearchElements/NodeModelSearchElement.cs

```diff
@@ -12,19 +12,25 @@ namespace Dynamo.Search.SearchElements
     public class NodeModelSearchElement : NodeModelSearchElementBase
     {
         private readonly Func<NodeModel> constructor;
+        private readonly string creationName;
 
+        public override string CreationName { get {return creationName;} }
+        
```

Comment:
yes, it will change what creationName we send to Reach in the library response, I think I have made changes that will take care of it in my save JSON PR to deal with the renamed nodes anyway, but I'm not sure, we'll need to inspect both the `nodeFactory` and the method `addNodeByNameAndPosition`. I think they'll both need modification to deal with this change.